src/algorithms/digi/LGADPulseGeneration.h

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Souvik Paul, Chun Yuen Tsang, Prithwish Tribedy
+// Special Acknowledgement: Kolja Kauder
+//
+// Convert energy deposition into ADC pulses
+// ADC pulses are assumed to follow the shape of landau function
+
+#pragma once
+
+#include <DDRec/CellIDPositionConverter.h>
+#include <algorithms/algorithm.h>
+#include <edm4hep/RawTimeSeriesCollection.h>
+#include <edm4hep/SimTrackerHitCollection.h>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "algorithms/digi/LGADPulseGenerationConfig.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+using LGADPulseGenerationAlgorithm =
+    algorithms::Algorithm<algorithms::Input<edm4hep::SimTrackerHitCollection>,
+                          algorithms::Output<edm4hep::RawTimeSeriesCollection>>;
+
+class PulseShape {
+public:
+  // return voltage in units of ADC value
+  // ranges from 0 to adc_range
+  // charge is in unit GeV, it's the EDep in the detector
+  virtual double Eval(double time, double hit_time, double charge) = 0;
+  virtual ~PulseShape() {};
+};
+
+// default pulse shape is Landau
+class LandauPulse : public PulseShape {
+public:
+  LandauPulse(double gain, double Vm, double sigma_analog, double adc_range);
+  double Eval(double time, double hit_time, double charge);
+private:
+  double m_gain;
+  double m_sigma_analog;
+  double m_scalingFactor;
+};
+
+
+class LGADPulseGeneration : public LGADPulseGenerationAlgorithm,
+                           public WithPodConfig<LGADPulseGenerationConfig> {
+// The key pair is <NO of 40 MHz cycle since t = 0, cellID>
+typedef std::unordered_map<dd4hep::rec::CellID, std::unordered_map<int, std::vector<double>>> AdcArray;
+
+public:
+  LGADPulseGeneration(std::string_view name, std::unique_ptr<PulseShape>&& pulse)
```

Comment:
```suggestion
  LGADPulseGeneration(std::string_view name)
```
Please stick to the interface. All configuration should come through the config. Ideally it's a POD type ready for (de-)serializtion, so if you need to support several shapes make a string key.